clang-tools-extra/docs/clang-tidy/checks/bugprone/null-check-after-dereference.rst

```diff
@@ -0,0 +1,83 @@
+.. title:: clang-tidy - bugprone-experimental-reverse-null
+
+bugprone-null-check-after-dereference
+=====================================
+
+*Note*: This check uses a flow-sensitive static analysis to produce its
+results. Therefore, it may be more resource intensive (RAM, CPU) than the
+average clang-tidy check.
+
+This check identifies potentially unsafe pointer null-checks,
+by finding cases where the pointer cannot be null at the location of the check.
+
+.. code-block:: c++
+
+   int f(int *ptr) {
+     *ptr = 20;
+     // ...
+     if (ptr) { // unsafe: pointer is checked after it has already been dereferenced.
+       return *ptr;
+     }
+     return 0;
+   }
+
+The check currently only supports C++ due to limitations in the data-flow framework. 
+A future version will add support for C code as well.
+
+Supported pointer operations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Pointer dereferences
+--------------------
+
+Pointer dot- and arrow-dereferences are supported. In the future, the checker could
+also be extended to find cases where a potentially-null pointer can be dereferenced.
+(This case is currently gives noisy results).
```

Comment:
A "potentially null pointer" in this case would mean that we can show a previous location where the pointer's value was definitely null. This is noisy because the check only knows definite pointer values in simpler cases, and thus this specific extension would have even more false negatives.
Removed the "In the future" section altogether.